src/cobra/flux_analysis/reaction.py

```diff
@@ -1,50 +1,52 @@
-# -*- coding: utf-8 -*-
-
-""" functions for analyzing / creating objective functions """
-
-from __future__ import absolute_import, division
+"""Provide functions for analyzing / creating objective functions."""
 
 from operator import attrgetter
+from typing import TYPE_CHECKING, Dict, Union
 from warnings import warn
 
-from cobra.core import Reaction
+from ..core import Reaction
+
+
+if TYPE_CHECKING:
+    from cobra import Model
 
 
-def assess(model, reaction, flux_coefficient_cutoff=0.001, solver=None):
-    """Assesses production capacity.
+def assess(
+    model: "Model",
+    reaction: Union[Reaction, str],
+    flux_coefficient_cutoff: float = 0.001,
+) -> Union[bool, Dict]:
```

Comment:
You are right. I missed that that #636 was meant to replace explicit passing of solver as an argument. I will revert this change and issue a deprecation warning for the argument.

Reply:
Actually, you're making a good argument **for** removing it right away 😄

Reply:
Having noticed no usage of it, I was tempted to remove it. But, I think based on your point, it's fair to issue a deprecation and then remove it so that it gets no user by surprise and helps us keep a track in the future.